### PR TITLE
feat: Measure time to join strategy [SNS-2154]

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

File: arroyo/processing/processor.py [hunk 1]
```diff
@@ -111,10 +112,18 @@ def _close_strategy() -> None:
...
                 try:
                     self.__processing_strategy.join(self.__join_timeout)
+                    self.__metrics_buffer.increment(
+                        ConsumerTiming.CONSUMER_JOIN_TIME, time.time() - start_join
+                    )
                     break
                 except InvalidMessage as e:
+                    self.__metrics_buffer.increment(
+                        ConsumerTiming.CONSUMER_JOIN_TIME, time.time() - start_join
+                    )
```
Comment (lynnagara) [inline arroyo/processing/processor.py:126]: should this be below line 127?

Comment (untitaker) [reply arroyo/processing/processor.py:126]: if we do that, then CONSUMER_DLQ_TIME overlaps with CONSUMER_JOIN_TIME

Comment (lynnagara) [reply arroyo/processing/processor.py:126]: Yeah. I'm worried the join time is understated if we don't. Knowing the actual total join time is important here, not just the ratio of join time to other time (as we want to know how close to the 5 second mark - when we start getting those rebalance timeout errors - we are).

Comment (lynnagara) [reply arroyo/processing/processor.py:126]: could be CONSUMER_JOIN_DLQ_TIME?

Comment (untitaker) [reply arroyo/processing/processor.py:126]: How about CONSUMER_SHUTDOWN_TIME is added as additional measurement? #215 already adds one that overlaps with join/poll timings

Comment (lynnagara) [reply arroyo/processing/processor.py:126]: Ignore that, separate CONSUMER_SHUTDOWN_TIME is better

**Review by lynnagara — APPROVED** [review]

**Pull request merged**